go/pkg/cas/upload.go

```diff
@@ -38,7 +39,14 @@ type UploadInput struct {
 	// Ignored if Path is not empty.
 	Content []byte
 
-	// TODO(nodir): add Predicate.
+	// PathExclude is a file/dir filter. If PathExclude is not nil and the
+	// absolute path of a file/dir match this regexp, then the file/dir is skipped.
+	// If the Path is a directory, then the filter is evaluated against each file
+	// in the subtree.
+	// See ErrSkip comments for more details on semantics.
```

Comment:
I'm not sure how ErrSkip is being used along PathExclude? Reading the code it seems that ErrSkip is exclusively being used with callback, which is (afaict) a completely separate thing from PathExclude.

Reply:
ErrSkip itself isn't used in PathExclude, but the semantics documented in ErrSkip is common to both. I figured I could "reuse" docs, but I can copy them too if you prefer.

Specifically I am referring to semantics around symlinks.